src/storage/storage_manager.cpp

```diff
@@ -146,13 +146,16 @@ void SingleFileStorageManager::LoadDatabase() {
 		// check if the WAL file exists
 		if (fs.FileExists(wal_path)) {
 			// replay the WAL
-			truncate_wal = WriteAheadLog::Replay(db, wal_path);
+			replay_status = WriteAheadLog::Replay(db, wal_path);
 		}
 	}
 	// initialize the WAL file
 	if (!read_only) {
 		wal = make_uniq<WriteAheadLog>(db, wal_path);
-		if (truncate_wal) {
+		if (replay_status.should_checkpoint) {
```

Comment:
I might be reading this wrong, but can't `should_checkpoint` be uninitialized here?
No default value for the variables is provided.

Reply:
Oh sorry - I think I forgot to initialize after removing the default values. Thanks for the catch!